packages/core/src/StateNode.ts

```diff
@@ -974,7 +978,7 @@ class StateNode<
               isInFinalState(transition.configuration, parentNode)
             )
           ) {
-            events.push(done(grandparent.id, grandparent.data));
```

Comment:
`grandparent.data` was always undefined anyway because only `.data` of final states has been copied onto state nodes:
https://github.com/davidkpiano/xstate/pull/1262/files#diff-f8963c788047c03637a3b59926c0d45dL376-R381

This also matches SCXML spec - those done events for parallel states do not hold any data.